src/pipecat/services/asyncai/tts.py

```diff
@@ -333,16 +332,14 @@ async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
                 await self._get_websocket().send(msg)
                 await self.start_tts_usage_metrics(text)
             except Exception as e:
-                logger.error(f"{self} exception: {e}")
-                yield ErrorFrame(error=f"{self} error: {e}")
+                await self.push_error(exception=e)
```

Comment:
@markbackman , regarding this kind of change, for everything inside an `AsyncGenerator`, instead of invoking `push_error` directly, I was always yielding all the frames, including the `ErrorFrame`.

And only inside the [process_generator](https://github.com/pipecat-ai/pipecat/blob/c6fcf58be4a63de1819e30e301f37cc5752c0d3f/src/pipecat/services/ai_service.py#L157), it is where we were actually pushing the `ErrorFrame`.

What do you think the pattern should be here? Because it feels more correct to always yield the frames to the generator instead of pushing them directly.